deeprank/features/atomicfeature_1.py

```diff
@@ -0,0 +1,703 @@
+# -*- coding: utf-8 -*-
+"""
+Spyder Editor
+
+This is a temporary script file.
+"""
+import os
+import warnings
+
+import numpy as np
+import pdb2sql
+
+from deeprank.features import FeatureClass
+from deeprank.config import logger
+from deeprank.operate.pdb import get_residue_contact_atom_pairs
+
+class AtomicFeature(FeatureClass):
+
+    def __init__(self, pdbfile, mutant, param_charge=None,
+                param_vdw=None, patch_file=None, contact_cutoff=8.5,
+                verbose=False):
+        """Compute the Coulomb, van der Waals interaction and charges around
+            the mutant position.
+
+        Args:
+
+            pdbfile (str): pdb file of the molecule
+
+            mutant: mutant position
+
+            param_charge (str): file name of the force field file
+                containing the charges e.g. protein-allhdg5.4_new.top.
+                Must be of the format:
+                * CYM  atom O   type=O      charge=-0.500 end
+                * ALA    atom N   type=NH1     charge=-0.570 end
+
+            param_vdw (str): file name of the force field containing
+                vdw parameters e.g. protein-allhdg5.4_new.param.
+                Must be of the format:
+                * NONBonded  CYAA    0.105   3.750       0.013    3.750
+                * NONBonded  CCIS    0.105   3.750       0.013    3.750
+
+            patch_file (str): file name of a valid patch file for
+                the parameters e.g. patch.top.
+                The way we handle the patching is very manual and
+                should be made more automatic.
+
+            contact_cutoff (float): the maximum distance in Å
+                between 2 contact atoms.
+
+            verbose (bool): print or not.
+
+        Examples:
+            >>> pdb = '1AK4_100w.pdb'
+            >>>
+            >>> # get the force field included in deeprank
+            >>> # if another FF has been used to compute the ref
+            >>> # change also this path to the correct one
+            >>> FF = pkg_resources.resource_filename(
+            >>>     'deeprank.features','') + '/forcefield/'
+            >>>
+            >>> # declare the feature calculator instance
+            >>> atfeat = AtomicFeature(pdb,
+            >>>    param_charge = FF + 'protein-allhdg5-4_new.top',
+            >>>    param_vdw    = FF + 'protein-allhdg5-4_new.param',
+            >>>    patch_file   = FF + 'patch.top')
+            >>>
+            >>> # assign parameters
+            >>> atfeat.assign_parameters()
+            >>>
+            >>> # only compute the pair interactions here
+            >>> atfeat.evaluate_pair_interaction(save_interactions=test_name)
+            >>>
+            >>> # close the db
+            >>> atfeat.sqldb._close()
+        """
+        super().__init__("Atomic")
+
+        # set a few things
+        self.pdbfile = pdbfile
```

Comment:
the path of the pdb file is actually included in the mutant object: https://github.com/DeepRank/deeprank/blob/efro-project/deeprank/models/mutant.py#L22
No need to input it as a separate argument.